### ISSUE TITLE
Marking your own expense as SPAM should'nt block the feature for your account

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#7159]
It currently does, but admins are sometimes just testing out the software.

**Comment by znarf** [comment]
Do we still have the (ugly alert) warning when doing so?

**Comment by Betree** [comment]
It's not ugly anymore, but yes. BTW the placeholder in this modal is not suited.

![image](https://github.com/opencollective/opencollective/assets/1556356/cfa6f99f-662f-4e1e-9d30-717cbd2f79a5)

**Comment by znarf** [comment]
Ok, but in a previous version we had a `window.alert` warning you that the account will be blocked etc ... Is that still existing somehow?

**Comment by Betree** [comment]
> Ok, but in a previous version we had a `window.alert` warning you that the account will be blocked etc ... Is that still existing somehow?

It's there in the screenshot I shared in the comment above:
> This will prevent the submitter account to post new expenses. Are you sure?

**Comment by znarf** [comment]
Indeed. I was surprised and distracted by the textarea, maybe because the warning was placed like a label for the textarea. Maybe we need to apply a different treatment for the warning with an icon and some yellow color.

**Comment by Betree** [comment]
It's also not clear by looking at it what the textarea stands for: a logging message only for the admins? A message sent to the contributor? A public comment?

A small label wouldn't hurt.

**Comment by MayaSatishRao** [comment]
@Betree I want to contribute to this issue and I want to understand it completely. Should the textarea in alert message not appear? And the appearance of the alert needs to be changed?

**Comment by MayaSatishRao** [comment]
@Betree 

> @Betree I want to contribute to this issue and I want to understand it completely. Should the textarea in alert message not appear? And the appearance of the alert needs to be changed?

@Betree ??

**Comment by Betree** [comment]
@MayaSatishRao please don't flood with comments; this is disrespectful towards the maintainers of this repo.

According to the comments above, we want to:
- Add a label on the textarea so people know what this message is for
- Change the placeholder to something more appropriate

To resolve the root cause of this issue, I would add to implement a conditional state for the modal where we don't let people mark their own expenses as SPAM and instead show a message like "You cannot mark your own expenses as SPAM".

**Comment by samkevin1** [comment]
@Betree Hey! I'm taking a look at this one but maybe you can clearify something for me. Inside an expense there is the `createdByAccount` object, and inside that object there's an ID (which I assume is the user/account ID). My idea was to use that ID and compare it with the ID of the `LoggedInUser`, and if those matched the user wouldn't be able to mark the expense as spam, but those IDs are not matching. Are they not both the user ID?

**Comment by Betree** [comment]
Hi @samkevin1, you're on the right path. The ID mismatch you're seeing is because `createdByAccount` is an `Account` (see that as "Profile") and `LoggedInUser` is a `User` from our legacy API. It's hard to guess from the outside.

The proper way to do this check should be something like:

```es6
const isSubmitter = expense.createdByAccount.legacyId === LoggedInUser.id;
```

You'll need to add the `legacyId` field in the relevant `createdByAccounts` in https://github.com/opencollective/opencollective-frontend/blob/dc23108d5c832699a7fb56c7bfee1aa4d06abcad/components/expenses/graphql/fragments.ts#L335.

**Comment by Betree** [comment]
> Hey @Betree ! Thanks for the explanation. Is comparing the collective IDs enough? I thought of comparing the user/account ids so we would know exactly which user created the expense.

Yes. Old names (GraphQLV1) were confusing: the `CollectiveId` means the id in the `Collectives` API table, which holds all the profile information (user profiles have an entry there with the `type=USER`). That's why we renamed them "Accounts" in our public GraphQL API. 

In GraphQLV2 terms, `LoggedInUser.CollectiveId` means the "Account ID of the logged in user".

**Comment by Betree** [comment]
Resolved by https://github.com/opencollective/opencollective-frontend/pull/10025
